src/Service/CloudFront/src/ValueObject/Paths.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace AsyncAws\CloudFront\ValueObject;
+
+use AsyncAws\Core\Exception\InvalidArgument;
+
+final class Paths
+{
+    /**
+     * The number of invalidation paths specified for the objects that you want to invalidate.
+     */
+    private $Quantity;
+
+    /**
+     * A complex type that contains a list of the paths that you want to invalidate.
+     */
+    private $Items;
+
+    /**
+     * @param array{
+     *   Quantity: int,
+     *   Items?: null|string[],
+     * } $input
+     */
+    public function __construct(array $input)
+    {
+        $this->Quantity = $input['Quantity'] ?? null;
```

Comment:
Is this correct? Quantity is required.

Reply:
I think this was okey at the time we had setters at the value objects.

Reply:
I think this is okey. You get a nice error message when this value object is being converted to a request.

Reply:
yes. people are free to construct their object with constructor or setter.

the required field are checked when building the request